src/cascadia/TerminalApp/WindowNameConverter.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "WindowNameConverter.h"
+
+#include "WindowIdConverter.g.cpp"
+#include "WindowNameConverter.g.cpp"
+#include <LibraryResources.h>
+
+using namespace winrt::Windows;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Text;
+
+namespace winrt::TerminalApp::implementation
+{
+    Foundation::IInspectable WindowIdConverter::Convert(Foundation::IInspectable const& value,
+                                                        Windows::UI::Xaml::Interop::TypeName const& /* targetType */,
+                                                        Foundation::IInspectable const& /* parameter */,
+                                                        hstring const& /* language */)
+    {
+        // Returns a string like "Window: 12345" for the window with ID 12345
+        const auto& id = winrt::unbox_value_or<uint64_t>(value, 0);
+        return winrt::box_value(fmt::format(std::wstring_view(RS_(L"WindowIdPrefix")), id));
+    }
+
+    Foundation::IInspectable WindowIdConverter::ConvertBack(Foundation::IInspectable const& /*value*/,

```

Comment:
This seems like something we should do with a getter (`WindowNameForDisplay()`, with a notification propagator) and just the _numerical ID getter_... like the about dialog's version number.

```
                    <Run x:Uid="AboutDialog_VersionLabel" />
                    <Run Text="{x:Bind ApplicationVersion}" />
```

That way, we don't need an entire converter just to produce a localized string.

(Also: localizing the entire string with the `:` and the `{}` is only necessary if we think the order of the words will change. In this case, it will not -- it is a field label ("Window") followed by a field value ("3")).. it produces undue localization burden to couple the number in with the name like this.

We can just name it "WindowToast_WindowIdLabel" and give it a comment that says "It is displayed as `Label: 10` or whatever"

Reply:
okeydokey.